datalad/distribution/publish.py

```diff
@@ -110,13 +110,11 @@ def has_diff(ds, refspec, remote, paths):
         # in which case we can do the same muuuch cheaper (see below)
         # if there were custom paths, we will look at the diff
         lgr.debug("Since paths provided, looking at diff")
-        return any(ds.diff(
-            path=within_ds_paths,
-            revision=remote_ref,
-            # only commited changes in this dataset
-            staged=False,
-            # consider only commited changes in subdataset
-            ignore_subdatasets='dirty')) > 0
+        return any(r["state"] != "clean"
+                   for r in ds.diff(path=within_ds_paths,
+                                    fr="HEAD",
+                                    to=remote_ref,
+                                    untracked="no"))
```

Comment:
I think this is a good conversion, but please double check.

Reply:
Looks good.